[stable/traefik] Upgrade traefik version to 1.3.4, fix toleration with 1.5 cluster and give traefik access to secrets for the k8s basic authentication

djsly opened the pull request:
if you guys could pick this up, it would be appreciated! :) 

@krancour @viglesiasce 

Thanks!

Commit:
update traefik version to 1.3.4

Commit:
Fixing the toleration support for 1.5.X k8s cluster

kachkaev commented:
@djsly can you also add this to your PR? https://github.com/kubernetes/charts/pull/1452 I'll close mine then.

Commit:
applying the following fix from upstream:
https://github.com/containous/traefik/commit/ca5bbab20a661b3b3810aeba84f2169322c88fd4

djsly commented:
@kachkaev : done! :) 
I have added your commit from #1452

krancour commented:
In the interest of tractability, it might be better for these three changes to each be their own PR. I know one of the features was (#1452) and seems to have been added into here because @kabakaev is eager.

If you need more convincing, two out of three of these changes I could LGTM right away, but one requires a little more research on my part to confirm it. By bundling the three together, the two that could be merged sooner get held up.

djsly commented:
Hello @krancour , one of the main reason I kept all three changes in three different commits was to help identifying the three changes seperatly. With the dependency of having to update the `Charts.yaml --> version: ` with every PR, it makes it that we often end up having to rebase depending on the order the PR are accepted. 

I will try to create multiple PR in the future.

krancour commented:
>  one of the main reason I kept all three changes in three different commits was to help identifying the three changes seperatly. With the dependency of having to update the Charts.yaml --> version: with every PR, it makes it that we often end up having to rebase depending on the order the PR are accepted.

Yep. Understood. Let me say, I appreciate that the three changes were three separate commits. That does make review easier. And the need to rebase depending on the merge order (to get version numbering right) is most definitely a pain. No argument there. But alas... the three commits are still bundled together in one PR which means the fates of all three changes are tied together until all three can be resolved (LGTM'ed or vetoed).

At any rate... I don't want to waste more of everyone's time than I already have bikeshedding about PRs. I'm just going to buckle down and look closer at the one change I wasn't sure about so that I can try to LGTM this whole thing.

krancour commented on stable/traefik/Chart.yaml:
```diff
@@ -1,7 +1,7 @@
 name: traefik
-version: 1.7.0
-appVersion: 1.3.1
+version: 1.8.0
```
Presumably, the delta between Traefik `1.3.1` and `1.3.3` is only bug fixes. If that's the case, the net effect on the chart is also only bug fixes. So shouldn't the new chart version be `1.7.1`?

djsly replied:
sure thing. I wasn't fully aware of the logic behind the chart `Version`. I agree that it should follow the version of the image. e.g. Major upgrade of `appVersion`  will mean a major upgrade of `version`

djsly replied:
Fixed

krancour commented:
This LGTM, but I'm not sure why the commits jumped from 3 to 6 after your recent fix. If you've got a minute, can you just clean this up? Maybe start a new local branch from master, cherry pick the three relevant changes and then force push from that local branch to this one? It will be cleaner, easier to follow history.

djsly commented:
ohhh sorry i must of rebased from the wrong branch...

djsly commented:
Fixed @krancour , back to 3 commits.

krancour commented:
Totally LGTM now. cc @viglesiasce @lachie83

kachkaev commented:
ping @viglesiasce @lachie83 :–)

kachkaev commented:
Ouch. The PR is still not merged after 2 weeks. Ping @prydonius @unguiculus :pray:

kachkaev commented:
@djsly looks like it's time to upgrade traefik to `1.3.4` :smile:

djsly commented:
@kachkaev done.
@krancour what is the process of getting those PR merged ? this is my firsts involvement with the public charts, I getting a feeling that we might be better sticking to local/<charts> when using helm even for public charts.

prydonius reviewed (changes requested):
Hi @djsly, apologies for the slow review. Not sure if I'm missing something but it doesn't look like the imageTag in values.yaml has been updated to 1.3.4, which means that this will stay on 1.3.1. https://github.com/djsly/charts/blob/ca9627cdcb28f72fc3ae22cf73820ab215d2f3ca/stable/traefik/values.yaml#L3

djsly commented:
@prydonius thanks... Sorry about that, I got confused by the `appVersion: 1.3.4` field in the Charts.yaml. good catch!

prydonius commented:
@djsly no problem, the appVersion is just for metadata (it doesn't necessarily relate to the image tags used, as there could be multiple images in a given application).

prydonius commented:
/ok-to-test

prydonius commented:
/test all

kachkaev commented:
It's probably because 1.3.4 is not yet in https://hub.docker.com/_/traefik/. It's in https://hub.docker.com/r/containous/traefik/ though. Explanation of the difference: https://github.com/containous/traefik-library-image/issues/3#issuecomment-318429169

The official upgrade should not take long, because the PR has already been merged: https://github.com/docker-library/official-images/pull/3243

prydonius commented:
@kachkaev thanks for the explanation. @djsly up to you whether you want to put this back to 1.3.3 to get this in now and update to 1.3.4 later in a separate PR, or just wait on this until 1.3.4 is available.

djsly commented:
I dont' mind waiting 1 extra day assuming @prydonius will be ready to issue a /retest once ready. me and @kachkaev can monitor for the release of 1.3.4

prydonius commented:
@djsly sgtm - I believe you also have the ability to issue a `/retest` or `/test all` :)

djsly commented:
sweet!

On Thu, Jul 27, 2017 at 3:25 PM, Adnan Abdulhussein <
notifications@github.com> wrote:

> @djsly <https://github.com/djsly> sgtm - I believe you also have the
> ability to issue a /retest or /test all :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/charts/pull/1485#issuecomment-318462016>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEwEKgk34a7cp6b_-odZ_vIICwaoDXQJks5sSOQigaJpZM4OXxxD>
> .
>

kachkaev commented:
1.3.4 is there: https://hub.docker.com/r/library/traefik/tags/  🎉

kachkaev commented:
/test all

kachkaev commented:
@prydonius your move I believe :–)

prydonius commented:
Thanks @djsly and @kachkaev!!

kachkaev commented:
I'll submit a tiny PR now to replace `secret` with `secrets`
